### PR TITLE
Add support for wildcards, message, defualtFile and DefaultPath for Choosers

### PR DESCRIPTION
**Pull request by gonzygonz**
here is an example how to use:

```
    my_cool_parser = GooeyParser(description=desc)

    my_cool_parser.add_argument("FileChooser", help=file_help_msg, widget="FileChooser",
                                gooey_options={
                                    'wildcard':
                                        "Comma separated file (*.csv)|*.csv|"
                                        "All files (*.*)|*.*",
                                    'message': "pick me"
                                }
                                )
    my_cool_parser.add_argument("DirectoryChooser", help=file_help_msg, widget="DirChooser",
                                gooey_options={
                                    'wildcard':
                                        "Comma separated file (*.csv)|*.csv|"
                                        "All files (*.*)|*.*",
                                    'message': "pick folder",
                                    'defaultPath': "c:/batch"
                                }
                                )
    my_cool_parser.add_argument("FileSaver", help=file_help_msg, widget="FileSaver",
                                gooey_options={
                                    'wildcard':
                                        "Comma separated file (*.csv)|*.csv|"
                                        "All files (*.*)|*.*",
                                    'message': "pick folder",
                                    'defaultPath': "c:/batch",
                                    'defaultFile': "def_file.csv"
                                }
                                )
    my_cool_parser.add_argument("MultiFileSaver", help=file_help_msg, widget="MultiFileChooser",
                                gooey_options={
                                    'wildcard':
                                        "Comma separated file (*.csv)|*.csv|"
                                        "All files (*.*)|*.*",
                                    'message': "pick folder",
                                    'defaultPath': "c:/batch",
                                    'defaultFile': "def_file.csv"
                                }
                                )
```
the widget example can also be update to use these new features. if you want i can update the example as well.

### CODE REVIEW

**Comment by gonzygonz** [comment]
forgot to mention - Fixes #363

**Comment by gonzygonz** [comment]
Hi,
any feedback on this PR?

File: gooey/gui/components/widgets/core/chooser.py [hunk 1]
```diff
@@ -58,37 +58,53 @@ def setValue(self, value):
     def getValue(self):
         return self.widget.getValue()
 
+    def getChooserOptions(self):
+        options = self.Parent._options
+        defaultFile = options['defaultFile'] if 'defaultFile' in options else ''
+        defaultPath = options['defaultPath'] if 'defaultPath' in options else ''
+        message = options['message'] if 'message' in options else ''
+        wildcard = options['wildcard'] if 'wildcard' in options else ''
+        return defaultFile, defaultPath ,message, wildcard
```
Comment (chriskiehl) [inline gooey/gui/components/widgets/core/chooser.py]: I'd argue that you can get by without this method -- you're trading glorious open key/val maps for closed positional semantics. 

The former is open to new Gooey options in the future, the latter is coupled to the current length and order of the return value -- meaning, a change to `getChooserOptions` would require a change to every one of its call sites due to the tuple unpacking. 

So, let's just use the `options` map directly. Python gives an API for accessing a map with fallback defaults on key miss which'll get rid of the verbose ternary expressions and unpacking awkwardness. So, things should end up looking, more or less, like this:

```python 
    def getDialog(self):
        return wx.FileDialog(
            self,
            message=options.get('message', _('open_file')),
            style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
            wildcard=options.get('wildcard', wx.FileSelectorDefaultWildcardStr))
```

**Comment by gonzygonz** [comment]
hi again
your suggestion to use get() is a great improvement indeed!
i fixed my change to contain that and also added defaultFile and added this to new multiDirChooser.
i am new to github so had some struggle with the rebase and all, i hope i didn't mess up the history or anything else that would make the merge difficult for you.

let me know if you need anything else (after the pull request is taken i hope to find time and adjust the example as well)

**Pull request closed without merging**